File: tests/php/example1.php
<?php
// Fonction DataFlowTest
function DataFlowTest($filePath, $test) {
    $filePath = 'example backward';
    $newPath = $filePath;
    $newPath = functionTest();

    // Vérifie si le fichier existe
    if (!file_exists($newPath)) {
        return 'File does not exist';
    } else {
        // Lis le contenu du fichier
        $content = file_get_contents($newPath);
        if ($content === false) {
            return 'Error reading file';
        }
        return $content;
    }

    $newPath = 'test';
}

// Fonction functionTest
function functionTest() {
    return 'example backward';
}

// Fonction TEST2
function TEST2($test) {
    $test = 'example testAAA';
    return $test;
}

// Fonction test
function test() {
    $filePath = 'example.txt';
    if ($filePath == '') {
        echo 'File does not exist\n';
    }

    $testStr = 'test';
    TEST2($filePathModified);

    $filePathModified = $filePathModified . '1';
    $test = 'test';
    $message = DataFlowTest($filePathModified, $test);

    echo $message . '\n';
}

// Fonction principale
function main() {
    $filePath = 'example backward';
    test();
}

main();
?>

File: tests/rb/example1.rb
# Fonction DataFlowTest
def DataFlowTest(filePath, test)
  filePath = "example backward"
  newPath = filePath
  newPath = functionTest()

  # Vérifie si le fichier existe
  if !File.exist?(newPath)
    return "File does not exist"
  else
    # Lis le contenu du fichier
    begin
      return File.read(newPath)
    rescue
      return "Error reading file"
    end
  end

  newPath = "test"
end

# Fonction functionTest
def functionTest
  "example backward"
end

# Fonction TEST2
def TEST2(test)
  test = "example testAAA"
  test
end

# Fonction test
def test
  filePath = "example.txt"
  if filePath.empty?
    puts "File does not exist"
  end

  testStr = "test"
  TEST2(filePathModified)

  filePathModified = filePathModified + "1"
  test= "test"
  message = DataFlowTest(filePathModified, test)

  puts message
end

# Fonction principale
def main
  filePath = "example backward"
  test()
end

main
